corehq/apps/userreports/static/userreports/js/report_config.js

```diff
@@ -0,0 +1,251 @@
+                    self.selectedColumns.unshift(column);
+                }
+            }
+            self.setIsFormatEnabled();
+            self.refreshPreview();
+        });
+
+        self.isFormatEnabled = ko.observable(false);
+        self.setIsFormatEnabled = function () {
+            var isFormatEnabled = self.isGroupByEnabled() && self.selectedGroupByName();
+            self.isFormatEnabled(isFormatEnabled);
+            // enable "Format" dropdown for each column that is not the group-by column.
+            _.each(self.selectedColumns(), function (column) {
+                column.isFormatEnabled(isFormatEnabled && column.name !== self.selectedGroupByName());
+                column.isGroupByColumn(isFormatEnabled && column.name === self.selectedGroupByName());
+            });
+        };
+
+        self.newColumnName = ko.observable('');
+
+        self.selectedChart = ko.observable('none');
+        self.selectedChart.subscribe(function (newValue) {
+            if (newValue === "none") {
+                self.groupByHeading("Group By");
+                self.groupByColumnStatus("Grouped By");
+                self.previewChart(false);
+            } else {
+                self.groupByHeading("Categories");
+                self.groupByColumnStatus("Category");
+                self.previewChart(true);
+                self.refreshPreview();
+            }
+        });
+
+        self.previewChart = ko.observable(false);
+
+        self.sum = function (array) {
+            var sum = 0;
+            try {
+                _.each(array, function (x) {
+                    if (!_.isNumber(x)) {
+                        throw new TypeError('"' + x + '" is not a number.');
+                    }
+                    sum += x;
+                });
+            } catch (err) {
+                return "---";
+            }
+            return sum;
+        };
+
+        self.avg = function (array) {
+            var sum = self.sum(array);
+            if (_.isNumber(sum)) {
+                return sum / array.length;
+            } else {
+                return "---";
+            }
+        };
+
+        self.count = function (array) {
+            return array.length;
+        };
+
+        self.refreshPreview = function (columns) {
+            columns = typeof columns !== "undefined" ? columns : self.selectedColumns();
+            $('#preview').hide();
+            if (columns.length === 0) {
+                return;  // Nothing to do.
+            }
+            $.ajax({
+                url: self.dataSourceUrl,
+                type: 'post',
+                contentType: 'application/json; charset=utf-8',
+                data: JSON.stringify({'columns': columns, 'aggregate': self.isFormatEnabled()}),
+                dataType: 'json',
+                success: self.renderChart,
+            });
+        };
+
+        self.renderChart = function (data) {
```

Comment:
This updates the table as well, right? "renderReportPreview" or something like that might make more sense as the function name.